docs/en/sql-reference/functions/type-conversion-functions.md

```diff
@@ -431,15 +427,51 @@ Result:
 └─────────────────────┘
 ```
 
+## reinterpret(x, T) {#type_conversion_function-reinterpret}
+
+Use the same source in-memory bytes sequence for `x` value and reinterpret it to destination type
+
+Query:
+```sql
+SELECT reinterpret(toInt8(-1), 'UInt8') as int_to_uint,
+    reinterpret(toInt8(1), 'Float32') as int_to_float,
+    reinterpret('1', 'UInt32') as string_to_int;
+```
+
+Result:
+
+```
+┌─int_to_uint─┬─int_to_float─┬─string_to_int─┐
+│         255 │        1e-45 │            49 │
+└─────────────┴──────────────┴───────────────┘
+```
+
 ## CAST(x, T) {#type_conversion_function-cast}
 
-Converts input value `x` to the `T` data type.
+Converts input value `x` to the `T` data type. Unlike to `reinterpret` function use external representation of `x` value. 
```

Comment:
"External representation" - that's unclear. We can say that the type conversion is performed in natural way.

Reply:
done thanks for suggestion, i will use it in future